modules/stochastic_tools/test/tests/reporters/ActiveLearningGP/tests

```diff
@@ -0,0 +1,59 @@
+[Tests]
+  issues = "#22280"
+  design = "ActiveLearningMonteCarloSampler.md ActiveLearningGPDecision.md ActiveLearningGaussianProcess.md"
+  [sampling]
+    requirement = 'The system shall be be able to perform Monte Carlo sampling with active learning with'
+    [SingleProc_SingleRow_Ufunction]
+      type = 'JSONDiff'
+      input = Main_adam.i
+      allow_test_objects = true
+      jsondiff = 'SingleProcSingleRow_Ufunction.json'
+      detail = 'a single row in the sampler with single processor using the U learning function;'
+    []
+    [MultipleProc_MultipleRow_Ufunction]
+      type = 'JSONDiff'
+      input = Main_adam.i
+      allow_test_objects = true
+      jsondiff = 'MultipleProcMultipleRow_Ufunction.json'
+      max_parallel = 3
+      min_parallel = 3
+      cli_args = "Samplers/mc/num_batch=2 Outputs/file_base='MultipleProcMultipleRow_Ufunction'"
+      detail = 'multiple rows in the sampler with multiple processors that are not equal to the rows using the U learning function;'
+    []
+    [COV_squared_exp]
+      type = 'JSONDiff'
+      input = Main_adam.i
+      allow_test_objects = true
+      jsondiff = 'COV_squared_exp.json'
+      cli_args = "Reporters/conditional/learning_function='COV' Reporters/conditional/learning_function_threshold=0.025 Outputs/file_base='COV_squared_exp'"
+      detail = 'a single row in the sampler with single processor using the COV learning function with GP using the squared exponential kernel;'
+    []
+    [SingleProc_SingleRow_Ufunction_tao]
+      type = 'JSONDiff'
+      input = Main_tao.i
+      allow_test_objects = true
+      jsondiff = 'Ufunction_tao.json'
+      rel_err = 1e-3
+      abs_zero = 1e-3
+      detail = 'a single row in the sampler with single processor using the COV learning function with GP using the squared exponential kernel and TAO optimization;'
+    []
+  []
+
+  [errors]
+    requirement = 'The system shall throw an error when '
+    [batch_with_tao]
```

Comment:
Aren't these two last tests already covered in previous work? (when you added Adam)

Reply:
Correct. I wanted to make sure these checks are performed for active learning too. Would it be too costly?

Reply:
I don't think so. I was just worried about test-duplication.